install/install.ps1

```diff
@@ -8,87 +8,196 @@ $script:ErrorActionPreference='Stop'
+
+    # A running instance of azureauth can cause installation to fail, so we try to kill any running instances first.
+    # We suppress taskkill output here because this is a best effort attempt and we don't want the user to see its output.
+    # Here, Get-Process is used to first determine whether there is an existing azureauth process. If there is, kill the existing process first.
+    $ProcessCheck = Get-Process -Name azureauth -ErrorAction SilentlyContinue -ErrorVariable ProcessError
+    if ($ProcessCheck -ne $null)
+    {
+        Write-Verbose "Stopping any currently running azureauth instances"
+        taskkill /f /im azureauth.exe 2>&1 | Out-Null
+
+        # After killing the process it is still possible for there there to be locks on the files it was using (including
+        # its own DLLs). The OS may take an indeterminate amount of time to clean those up, but so far we've observed 1
+        # second to be enough.
+        Start-Sleep -Seconds 1
     }
-    Set-ItemProperty -Path $registryPath -Name PATH -Value $newPath
+
+    if (Test-Path -Path $extractedDirectory) {
+        Write-Verbose "Removing pre-existing extracted directory at ${extractedDirectory}"
+        Remove-Item -Force -Recurse $extractedDirectory
+    }
+
+    Write-Verbose "Extracting ${zipFile} to ${extractedDirectory}"
+    Add-Type -AssemblyName System.IO.Compression.FileSystem
+    [System.IO.Compression.ZipFile]::ExtractToDirectory($zipFile, $azureauthDirectory)
+
+##TODO - Remove the latestDirectory related code
+#    if (Test-Path -Path $latestDirectory) {
+#        Write-Verbose "Removing pre-existing latest directory at ${latestDirectory}"
+        
+        # We use IO.Directory::Delete instead of Remove-Item because on Windows Server 2012 with PowerShell 4.0 the latter will not work.
+#        [IO.Directory]::Delete($latestDirectory)
+#    }
+
+    # We use a directory junction here because not all Windows users will have permissions to create a symlink.
+    # We create this junction with cmd.exe's mklink because it has a stable interface across all active versions of Windows and Windows Server,
+    # while PowerShell's New-Item has breaking changes and doesn't have the -Target param in 4.0 (the default PowerShell on Win Server 2012).
+
+#    Write-Verbose "Linking ${latestDirectory} to ${extractedDirectory}"
+#    cmd.exe /Q /C "mklink /J `"$latestDirectory`" `"$extractedDirectory`"" > $null
+#    if (!$?) {
+#        Write-Error "Linking failed!"
+#    }
+
+    Write-Verbose "Removing ${zipFile}"
+    Remove-Item -Force $zipFile
+
+    # Permanently add the latest directory to the current user's $PATH (if it's not already there).
+    # Note that this will only take effect when a new terminal is started.
+    $registryPath = 'Registry::HKEY_CURRENT_USER\Environment'
+    $currentPath = (Get-ItemProperty -Path $registryPath -Name PATH -ErrorAction SilentlyContinue).Path
+    if ($currentPath -NotMatch 'AzureAuth') {

```

Comment:
In disucssion we noted how the curernt if block basically does what we want in this case where we do want to update the path but azureauth.exe isn't on it yet. Something I did not call out however, is how in our future world, `AzureAuth` itself may never be in the path. Since the directory env var could be custom, and the directory we install is going to change to only be the version, and not the full release asset name.